examples/step-17/step-17.cc

```diff
@@ -282,58 +293,97 @@ namespace Step17
+  // processes, commonly called "partitioning the mesh", is done by
+  // assigning a @ref GlossSubdomainId "subdomain id" to each cell. We
+  // do so by calling into the METIS library that does this in a very
+  // efficient way, trying to minimize the number of nodes on the
+  // interfaces between subdomains. Rather than trying to call METIS
+  // directly, we do this by calling the
+  // GridTools::partition_triangulation() function that does this at a
+  // much higher level of programming.
+  //
+  // @note As mentioned in the introduction, we can avoid this manual
+  //   partitioning step if we used the parallel::shared::Triangulation
+  //   class for the triangulation object instead (as we do in step-18).
+  //   That class does, in essence, everything a regular triangulation
+  //   does, but it then also automatically partitions the mesh after
+  //   every mesh creation or refinement operation.
+  //
+  // Following partitioning, we need to enumerate all degrees of
+  // freedom as usual.  However, we would like to enumerate the
+  // degrees of freedom in a way so that all degrees of freedom
+  // associated with cells in subdomain zero (which resides on process
+  // zero) come before all DoFs associated with cells on subdomain
+  // one, before those on cells on process two, and so on. We need
+  // this since we have to split the global vectors for right hand
+  // side and solution, as well as the matrix into contiguous chunks
+  // of rows that live on each of the processors, and we will want to
+  // do this in a way that requires minimal communication. This
+  // particular enumeration can be obtained by re-ordering degrees of
+  // freedom indices using DoFRenumbering::subdomain_wise().
+  //
+  // The final step of this initial setup is that we get ourselves a
+  // variable that indicates how many degrees of freedom the current
+  // process is responsible for. (This will, in general, be less than
+  // <code>fe.dofs_per_cell</code> times the number of cells the
+  // current process owns because some degrees of freedom live on
+  // interfaces between subdomains, and are consequently only owned by
+  // one of the processes adjacent to this interface.)
+  //
+  // Before we move on, let us recall a fact already discussed in the
+  // introduction: The triangulation we use here is replicated across
+  // all processes, and each process has a complete copy of the entire
+  // triangulation, with all cells. Partitioning only provides a way
+  // to identify which cells out of all each process "owns", but it
+  // knows everything about all of them. Likewise, the DoFHandler
+  // object knows everything about every cell, in particular the
+  // degrees of freedom that live on each cell, whether it is one that
+  // the current process owns or not. This can not scale to large
+  // problems because eventually just storing on every process the
+  // entire mesh and everything that is associated with it, will
```

Comment:
Another grammatical nitpick: I believe there should be a comma after `entire mesh`.